source/simulator/assemblers/stokes.cc

```diff
@@ -335,8 +335,11 @@ namespace aspect
   {
     // assemble RHS of:
     //  - div u = 1/rho * drho/dp rho * g * u
-    Assert(parameters.formulation_mass_conservation ==
-           Parameters<dim>::FormulationMassConservation::isothermal_compression,
+    Assert((parameters.formulation_mass_conservation ==
+            Parameters<dim>::FormulationMassConservation::isothermal_compression) ||
+           ((parameters.formulation_mass_conservation ==
+             Parameters<dim>::FormulationMassConservation::ask_material_model) &&
```

Comment:
I think the code would be much easier if we replace ask_material_model at initialization time (or if this is too early in your validation routine).